module/assets/helper.js

```diff
@@ -0,0 +1,1075 @@
+                    );
+                  }
+                }
+              }
+              //Queries to update owner in programs Collections for ownership transfer
+              let programFilter = {
+                owner: reqData.fromUserProfile.userId,
+              };
+              let updatePrograms = {
+                $set: {
+                  owner: reqData.toUserProfile.userId,
+                },
+              };
+              let updatePartialPrograms = await programsHelper.updateMany(
+                programFilter,
+                updatePrograms
+              );
+              updatePartialPrograms
+                ? (updateUserAssetDataResult = true)
+                : (updateUserAssetDataResult = false);
+            }
+          } else {
+            //condition for if there is an assetInformation in ownershipTransferEvent
+
+            let typeOfAssetsToMove = reqData.assetInformation.objectType;
+            if (
+              reqData.toUserProfile.roles.includes(
+                constants.common.CONTENT_CREATOR
+              ) &&
+              typeOfAssetsToMove === constants.common.SOULTION
+            ) {
+              //match and Set Queries for solutions partial Transfer
+              let solutionFilter = {
+                author: reqData.fromUserProfile.userId,
+                _id: new ObjectId(reqData.assetInformation.identifier),
+              };
+              let updateSolutions = {
+                $set: {
+                  author: reqData.toUserProfile.userId,
+                  creator: reqData.toUserProfile.firstName,
+                },
+              };
+              let solutionLicenseFilter = {
+                author: reqData.fromUserProfile.userId,
+                license: { $exists: true, $ne: "" },
+              };
+              let updateSolutionsLicense = {
+                $set: {
+                  "license.author": reqData.toUserProfile.firstName,
+                  "license.creator": reqData.toUserProfile.firstName,
+                },
+              };
+
+              let updateUserSolutions = [
+                solutionsHelper.updateMany(solutionFilter, updateSolutions),
+                solutionsHelper.updateMany(
+                  solutionLicenseFilter,
+                  updateSolutionsLicense
+                ),
+              ];
+              let updatedOneToOneTransferSolution = await Promise.all(
+                updateUserSolutions
+              );
+              updatedOneToOneTransferSolution
+                ? (updateUserAssetDataResult = true)
+                : (updateUserAssetDataResult = false);
+            }
+
+            if (
+              reqData.toUserProfile.roles.includes(
+                constants.common.PROGRAM_MANAGER
+              ) ||
+              (reqData.toUserProfile.roles.includes(
+                constants.common.PROGRAM_DESIGNER
+              ) &&
+                typeOfAssetsToMove === constants.common.PROGRAM)
```

Comment:
@praveenKDass Does this mean that I can transfer a program manager program to another user which has a program designer role?